libraries/precisoUtils/bidUtils.js

```diff
@@ -96,8 +77,161 @@ export function onBidWon(bid) {
+      ver: '1.2'
+    };
+    return {
+      request: JSON.stringify(request)
+    }
+  }
+}
+
+function mapBanner(slot) {
+  if (slot.mediaTypes.banner) {
+    let format = (slot.mediaTypes.banner.sizes || slot.sizes).map(size => {
+      return { w: size[0], h: size[1] }
+    });
+
+    return {
+      format
+    }
+  }
+}
+
+export function buildBidResponse(serverResponse) {
+  const responseBody = serverResponse.body;
+  const bids = [];
+  responseBody.seatbid.forEach(seat => {
+    seat.bid.forEach(serverBid => {
+      if (!serverBid.price) {
+        return;
+      }
+      if (serverBid.adm.indexOf('{') === 0) {
+        bids.push({
+          requestId: serverBid.impid,
+          mediaType: NATIVE,
+          cpm: serverBid.price,
+          creativeId: serverBid.crid,
+          width: 1,
+          height: 1,
+          ttl: 300,
+          meta: {
+            advertiserDomains: serverBid.adomain
+          },
+          netRevenue: true,
+          currency: 'USD',
+          native: interpretNativeAd(macroReplace(serverBid.adm, serverBid.price)),
+        });
+      } else {
+        bids.push({
+          requestId: serverBid.impid,
+          cpm: serverBid.price,
+          width: serverBid.w,
+          height: serverBid.h,
+          creativeId: serverBid.crid,
+          ad: macroReplace(serverBid.adm, serverBid.price),
+          currency: 'USD',
+          netRevenue: true,
+          ttl: 300,
+          meta: {
+            advertiserDomains: serverBid.adomain || '',
+          },
+        });
+      }
+    })
+  });
+  return bids;
+}
+
+function interpretNativeAd(adm) {
+  try {
+    const native = JSON.parse(adm).native;
+    if (native) {
+      const result = {
+        clickUrl: encodeURI(native.link.url),
+        impressionTrackers: native.eventtrackers[0].url,
+        clickTrackers: native.link.clicktrackers[0]
+
+      };
+      native.assets.forEach(asset => {
+        switch (asset.id) {
+          case OPENRTB.NATIVE.ASSET_ID.TITLE:
+            result.title = asset.title.text;
+            break;
+          case OPENRTB.NATIVE.ASSET_ID.IMAGE:
+            result.image = {
+              url: encodeURI(asset.img.url),
+              width: asset.img.w || 300,
+              height: asset.img.h || 250
+            };
+            break;
+          case OPENRTB.NATIVE.ASSET_ID.ICON:
+            result.icon = {
+              url: encodeURI(asset.img.url),
+              width: asset.img.w || 10,
+              height: asset.img.h || 10
+            };
+            break;
+          case OPENRTB.NATIVE.ASSET_ID.DATA:
+            result.body = asset.data.value;
+            break;
+          case OPENRTB.NATIVE.ASSET_ID.SPONSORED:
+            result.sponsoredBy = asset.data.value;
+            break;
+          case OPENRTB.NATIVE.ASSET_ID.CTA:
+            result.cta = asset.data.value;
+            break;
+        }
+      });
+      return result;
+    } else {
+      return
```

Comment:
A native bid w/o assets should be filtered out of your returned bid responses, could you add this?